noir-projects/aztec-nr/aztec/src/state_vars/single_use_claim.nr

```diff
@@ -0,0 +1,266 @@
+/// }
+/// ```
+pub struct SingleUseClaim<ClaimIdentifier, Context> {
+    context: Context,
+    storage_slot: Field,
+}
+
+/// Represents the SingleUseClaim scoped to an owner address.
+/// See: SingleUseClaim::at.
+pub struct SingleUseClaimWithOwner<ClaimIdentifier, Context> {
+    context: Context,
+    storage_slot: Field,
+    owner: AztecAddress,
+}
+
+// Private storage slots are not really 'slots' but rather a value in the nullifier hash preimage,
+// so there is no notion of a value spilling over multiple slots.
+// For this reason SingleUseClaim (just as any other private state variable) needs
+// just one slot to be reserved, regardless of what it stores.
+impl<ClaimIdentifier, Context> HasStorageSlot<1> for SingleUseClaim<ClaimIdentifier, Context> {
+    fn get_storage_slot(self) -> Field {
+        self.storage_slot
+    }
+}
+
+impl<ClaimIdentifier, Context> SingleUseClaim<ClaimIdentifier, Context> {
+    /// Initializes a new SingleUseClaim state variable.
+    /// This function is usually automatically called within the `#[storage]` macro.
+    /// You typically don't need to call this directly when writing smart contracts.
+    ///
+    /// # Arguments
+    ///
+    /// * `context` - One of `PrivateContext`/`PublicContext`/`UtilityContext`. The
+    ///               Context determines which methods of this struct will be made
+    ///               available to the calling smart contract function.
+    /// * `storage_slot` - A unique identifier for this state variable within the
+    ///                    contract. The nullifier for this SingleUseClaim
+    ///                    state variable will have this `storage_slot`.
+    ///                    Usually, the `#[storage]` macro will determine an
+    ///                    appropriate `storage_slot` automatically. A smart contract
+    ///                    dev shouldn't have to worry about this, as it's managed
+    ///                    behind the scenes.
+    ///
+    /// docs:start:new
+    pub fn new(context: Context, storage_slot: Field) -> Self {
+        assert(storage_slot != 0, "Storage slot 0 not allowed. Storage slots must start from 1.");
+        Self { context, storage_slot }
+    }
+
+    /// Returns a SingleUseClaimWithOwner instance, which is the original
+    /// SingleUseClaim state variable scoped to the given owner address.
+    ///
+    /// The SingleUseClaim state can then be written by calling `claim` on it, or read
+    /// via `has_claimed` (utility function), or `assert_claimed` (private function).
+    ///
+    /// # Arguments
+    ///
+    /// * `owner` - The AztecAddress to which we scope this variable.
+    ///
+    /// # Returns
+    ///
+    /// * `SingleUseClaimWithOwner` - The original `SingleUseClaimWithOwner` augmented with an `owner` attribute.
+    ///         You can then call methods like `.claim()`, `.assert_claimed()`, `.has_claimed()`, etc. which
+    ///         are scoped to the given owner address.
```

Comment:
We don't need to follow a rigid structure with `Arguments` and `Returns` sections, e.g. `owner` in this case has already been described.